includes/Modules/Sign_In_With_Google.php

```diff
@@ -221,33 +392,54 @@ private function render_signin_button() {
 			return;
 		}
 
-		$redirect_url = add_query_arg( 'action', 'google_auth', wp_login_url() );
-		if ( substr( $redirect_url, 0, 5 ) !== 'https' ) {
+		$login_uri = add_query_arg( 'action', 'google_auth', wp_login_url() );
+		if ( substr( $login_uri, 0, 5 ) !== 'https' ) {
 			return;
 		}
 
+		$redirect_to = $this->context->input()->filter( INPUT_GET, 'redirect_to' );
+		$redirect_to = trim( $redirect_to );
+
 		// Render the Sign in with Google button and related inline styles.
 		?>
-<!-- <?php echo esc_html__( 'Sign in with Google button added by Site Kit', 'google-site-kit' ); ?> -->
+<!-- Sign in with Google button added by Site Kit -->
```

Comment:
WDYT about extracting this to its own class similar to module `Web_Tag`s?

Reply:
I don't think there is a need for it. This module is pretty small and there is nothing else that we will need to add to this module. Let's keep it all together. We can always split it in the future if a need be.